src/sources/font_preferences.ts

```diff
@@ -52,13 +52,14 @@ export const presets: Record<string, Preset> = {
  * The "min" and the "mono" (only on Windows) value may change when the page is zoomed in Firefox 87.
  */
 export default function getFontPreferences(): Promise<Record<string, number>> {
-  return withNaturalFonts((document, container) => {
+  return withNaturalFonts(async (document, body) => {
     const elements: Record<string, HTMLElement> = {}
     const sizes: Record<string, number> = {}
+    const container = document.createElement('div')
 
     // First create all elements to measure. If the DOM steps below are done in a single cycle,
     // browser will alternate tree modification and layout reading, that is very slow.
-    for (const key of Object.keys(presets)) {
+    await mapWithBreaks(Object.keys(presets), async (key) => {
```

Comment:
Are you sure `mapWithBreaks` is necessary? There are only 7 presets, and the actions inside are very small.

The callback is not necessary to be `async`, there is no `await` inside, and `mapWithBreaks` is made for synchronous callbacks. The `async` only makes browsers create unnecessary promises at runtime.

Reply:
I checked both cases, `mapWithBrakes` never reaches `await wait(0)` in tests 

So perhaps durations are only affected by moving sync operations to microtasks